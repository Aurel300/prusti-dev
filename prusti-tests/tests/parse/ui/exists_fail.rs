// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Pno_verify=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"
// normalize-stdout-test: "\[[a-z0-9]{4}\]::" -> "[$(CRATE_ID)]::"

use prusti_contracts::*;

#[requires(exists)]
fn test1() {}

#[requires(exists())]
fn test2() {}

#[requires(exists(|))]
fn test3() {}

#[requires(exists(||) 1+1)]
fn test4() {}

#[requires(exists(||) || exists(||))]
fn test5() {}

#[requires(exists(|| 1+1 == 1+1, triggers=[1]))]
fn test6() {}

#[requires(exists(|| true, triggers=[(1,2), 1]))]
fn test7() {}

#[requires(exists(|| true, triggers=1))]
fn test8() {}

#[requires(exists(||))]
fn test9() {}

#[requires(exists(|| 1+1 == 1+1))]
fn test10() {}

#[requires(exists(||, triggers=[]))]
fn test11() {}

#[requires(exists(|| 1+1 == 1+1, triggers=[(1,)]))]
fn test12() {}

fn main() {}
