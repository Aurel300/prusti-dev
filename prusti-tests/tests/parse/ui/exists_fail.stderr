error: expected parenthesized expression after quantifier
 --> $DIR/exists_fail.rs:8:12
  |
8 | #[requires(exists)]
  |            ^^^^^^

error: expected quantifier body
  --> $DIR/exists_fail.rs:11:12
   |
11 | #[requires(exists())]
   |            ^^^^^^

error: expected quantifier body
  --> $DIR/exists_fail.rs:14:12
   |
14 | #[requires(exists(|))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:17:12
   |
17 | #[requires(exists(||) 1+1)]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:20:12
   |
20 | #[requires(exists(||) || exists(||))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:23:12
   |
23 | #[requires(exists(|| 1+1 == 1+1, triggers=[1]))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:26:12
   |
26 | #[requires(exists(|| true, triggers=[(1,2), 1]))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:29:12
   |
29 | #[requires(exists(|| true, triggers=1))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:32:12
   |
32 | #[requires(exists(||))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:35:12
   |
35 | #[requires(exists(|| 1+1 == 1+1))]
   |            ^^^^^^

error: expected quantifier body
  --> $DIR/exists_fail.rs:38:12
   |
38 | #[requires(exists(||, triggers=[]))]
   |            ^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/exists_fail.rs:41:12
   |
41 | #[requires(exists(|| 1+1 == 1+1, triggers=[(1,)]))]
   |            ^^^^^^

error: aborting due to 12 previous errors

